orchagent/portsorch.cpp

```diff
@@ -2732,9 +2732,11 @@ void PortsOrch::doPortTask(Consumer &consumer)
                 continue;
             }
 
-            if (!gBufferOrch->isPortReady(alias))
+            if (!gBufferOrch->isPortReady(alias) && (admin_status == "up"))
```

Comment:
as discussed, why is this check needed?  BUFFER_PG and BUFFER_QUEUE tables only include active ports and isPortReady() will return true for admin down ports

Reply:
I understood the confusion.
In `isPortReady`, a port can be ready if one of the following conditions is satisfied:
- it's not in the list `m_port_ready_list_ref`, meaning there is not any `BUFFER_PG` item configured on the port. The `m_port_ready_list_ref` contains all the keys configured in `BUFFER_PG`
- it's in `m_ready_list`, meaning there are some configurations in the above table and all items in `BUFFER_PG` have been deployed to SAI.

Now that no buffer PG will be deployed to SAI if the port is admin down, which means it won't be `ready` until the port is admin up, we need to handle the case by testing whether the admin_status is `up`.
During the discussion, the `m_ready_list` wasn't found in the code somehow.

Reply:
I have some concerns regarding the lossless profiles. Currently they get applied to even admin down ports. With this check, we end up removing that port from pendingPortSet even if the profile hasn't been pushed down to SAI

Reply:
Fixed.